core/src/main/java/org/infinispan/remoting/inboundhandler/BasePerCacheInboundInvocationHandler.java

```diff
@@ -123,83 +133,27 @@ protected final void handleRunnable(BlockingRunnable runnable, boolean onExecuto
       }
    }
 
+   protected final boolean isCommandSentBeforeFirstTopology(int commandTopologyId) {
+      if (0 <= commandTopologyId && commandTopologyId < stateTransferManager.getFirstTopologyAsMember()) {
```

Comment:
Doesn't each invocation handler check `isCommandSentBeforeFirstTopology()` before executing the command?